src/backend/app/auth/roles.py

```diff
@@ -334,3 +334,60 @@ async def mapper(
         current_user,
         ProjectRole.MAPPER,
     )
+
+
+async def project_contributors(
+    project_user: Annotated[ProjectUserDict, Depends(mapper)],
+    db: Annotated[Connection, Depends(db_conn)],
+    current_user: Annotated[AuthUser, Depends(login_required)],
+) -> bool:
+    """A contributor to a specific project."""
+    user_id = current_user.id
+    project = project_user["project"]
+    project_id = project.id
+    org_id = project.organisation_id
+
+    query = """
+        SELECT
+            EXISTS (
+                SELECT 1 FROM projects
+                WHERE author_id = %(user_id)s
+            ) OR
+            EXISTS (
```

Comment:
I think it will be good to use a `CASE` statement like we do in check_access?  first checking higher roles followed accordingly.